server/src/main/java/org/apache/kafka/server/share/fetch/PartitionMaxBytesStrategy.java

```diff
@@ -58,10 +66,47 @@ static PartitionMaxBytesStrategy type(StrategyType type) {
     private static LinkedHashMap<TopicIdPartition, Integer> uniformPartitionMaxBytes(int requestMaxBytes, Set<TopicIdPartition> partitions, int acquiredPartitionsSize) {
         checkValidArguments(requestMaxBytes, partitions, acquiredPartitionsSize);
         LinkedHashMap<TopicIdPartition, Integer> partitionMaxBytes = new LinkedHashMap<>();
-        partitions.forEach(partition -> partitionMaxBytes.put(partition, requestMaxBytes / acquiredPartitionsSize));
+        if (requestMaxBytes >= acquiredPartitionsSize) {
+            // Case 1: requestMaxBytes can be evenly distributed within partitions. If there is extra bytes left post
+            // dividing it uniformly, assign it randomly to any one of the partitions.
+            partitions.forEach(partition -> partitionMaxBytes.put(partition, requestMaxBytes / acquiredPartitionsSize));
+            if (requestMaxBytes % acquiredPartitionsSize != 0) {
+                TopicIdPartition randomPartition = selectPartitionRandomly(partitionMaxBytes);
+                partitionMaxBytes.put(randomPartition,
+                    (requestMaxBytes / acquiredPartitionsSize) + (requestMaxBytes % acquiredPartitionsSize));
```

Comment:
nit: Though the calculation is trivial but not required to be repeated i.e. `requestMaxBytes / acquiredPartitionsSize` rather fetch the existing and add.

If you don't like it then ignore the suggestion.

```suggestion
            int remainingBytes = requestMaxBytes % acquiredPartitionsSize;
            if (remainingBytes != 0) {
                TopicIdPartition randomPartition = selectPartitionRandomly(partitionMaxBytes);
                partitionMaxBytes.put(randomPartition,
                    partitionMaxBytes.get(randomPartition) + remainingBytes);
```